src/controllers/cart.js

```diff
@@ -0,0 +1,111 @@
+const Cart = require('../models/cart');
+const CustomError = require('../utils/customError');
+
+exports.getCart = async (req, res, next) => {
+  try {
+    const cart = await Cart.findOne({ customerId: req.user.id });
+    if (!cart) return next(new CustomError('cart not found', 404));
+    return res.status(200).json(cart);
+  } catch (err) {
+    return next(new CustomError(err.message, 500));
+  }
+};
+
+exports.postCart = async (req, res, next) => {
+  try {
+    let cart = await Cart.findOne({ customerId: req.user.id });
+    if (cart) return next(new CustomError('cart already exist', 300));
+    cart = new Cart({ customerId: req.user.id });
+    await cart.save();
+    return res.status(201).json(cart);
+  } catch (err) {
+    return next(new CustomError(err.message, 500));
+  }
+};
+
+exports.deleteCart = async (req, res) => {
+  try {
+    Cart.deleteOne({ customerId: req.user.id });
+    res.status(204).json({
+      message: 'the cart was deleted',
+    });
+  } catch (err) {
+    res.status(500).json({ message: 'Failed to delete cart', error: err });
+  }
+};
+
+exports.postCartItemByDishId = async (req, res, next) => {
+  try {
+    const cart = await Cart.findOne({ customerId: req.user.id });
+    if (!cart) return next(new CustomError('cart not found', 404));
+
+    let cartItem = await Cart.findOne({
+      customerId: req.user.id,
+      'cartItems.dishId': req.params.id,
+    });
+    if (cartItem) return next(new CustomError('item already in cart', 300));
+
+    cartItem = { dishId: req.params.id, quantity: 1 };
```

Comment:
Why quantity is 1, what if the customer wanted quantity to be 3
and quantity already has a default value (1) in case the customer didn't specify the quantity, check cart model